### PR TITLE
MTM-59369: Hotfixes for automation of changelog dates

### PR DESCRIPTION
**Pull request by shaikhahme**
1. Now checks if a PR to update dates is already exists, only creates a new PR if PR doesn't already exist.
2. Converts version to semantic for processing and back to non-semantic before writing to changelogs

### CODE REVIEW

File: add-release-dates/index.ts [hunk 1]
```diff
@@ -56,6 +53,8 @@ async function processFile(filePath: string): Promise<boolean> {
     );
     return false;
   }
+
+  data.version = convertVersionFormat(data.version) // converts from semantic semantic to non-semantic version format
```
Comment (reey) [inline add-release-dates/index.ts]: now we would convert previously valid semver to a four digit version number.. We only want to have the initial version number back here.. if it was previously valid semver, it should stay like that..
So maybe just store the initial version number in a variable in line 37 and then afterwards just set `data.version` to the initial value..

Comment (shaikhahme) [reply add-release-dates/index.ts]: resolved in d1e7f030c7a8e43aa4cbb5855a9e7d11736e392a.

File: add-release-dates/index.ts [hunk 2]
```diff
@@ -34,20 +34,17 @@ async function processFile(filePath: string): Promise<boolean> {
     console.warn("No version set in: ", filePath, "Skipping..");
     return false;
   }
+  originalVersion = data.version;
```
Comment (reey) [inline add-release-dates/index.ts]:
```suggestion
  const originalVersion = data.version;
```

File: add-release-dates/index.ts [hunk 3]
```diff
@@ -56,6 +53,8 @@ async function processFile(filePath: string): Promise<boolean> {
     );
     return false;
   }
+
+  data.version = originalVersion
```
Comment (reey) [inline add-release-dates/index.ts]:
```suggestion
  data.version = originalVersion;
```

**Review by jakubsobolewskisag — APPROVED** [review]

**Review by reey — APPROVED** [review]

**Review by sourajitsaha17 — APPROVED** [review]

**Comment by BeateRixen** [comment]
@shaikhahme @eliasweingaertner @reey Can this PR be merged so that we can proceed with the open PRs which add dates to various components? There are quite some change log entries that need to get published.

**Review by BeateRixen — APPROVED** [review]

**Comment by BeateRixen** [comment]
> @shaikhahme @eliasweingaertner @reey Can this PR be merged so that we can proceed with the open PRs which add dates to various components? There are quite some change log entries that need to get published.

I can of course also add the dates manually for now if the automation is still work in progress.

**Pull request merged**